### PR TITLE
Move from shurcooL/VFSgen to go embed.

### PR DESCRIPTION
**Pull request by kishen-v**
Changes introduced:

1. Eliminates dependency from `"github.com/shurcooL/vfsgen"` by using `go embed` directive.
2. The README.md has been updated with the required content, on steps to build the bin.
3. The files related to asset generation and the /hack/build.sh have been removed.
4. Assets for plugin have been moved from `kubetest2-plugins/data/data/*` to `kubetest2-plugins/data/*`
5. Update GitHub actions to execute `git submodule update --init` before `go build ./...`

As the repository now does not contain the assets, `git submodule update --init` needs to be executred before plugin installation. Upon execution, the `k8s-ansible` repository is cloned to the preset path, which can then be embedded during build phase.

- The periodic test-jobs which use the plugin are now updated to execute `git submodule update --init` as a step during the execution phase.  

- The presubmit job that validates the .tf file in this repository is also updated to point to the new path that contains the .tf files. PR : https://github.com/ppc64le-cloud/test-infra/pull/358

### CODE REVIEW

File: data/unpack.go [hunk 1]
```diff
@@ -21,24 +31,27 @@ func Unpack(base string, uri string) (err error) {
 	}
 
 	if info.IsDir() {
-		os.Mkdir(base, 0777)
-		children, err := file.Readdir(0)
+		err := os.MkdirAll(extractPath, 0777)
```
Comment (kishen-v) [inline data/unpack.go:34]: Switched to using `os.MkdirAll`, to suppress error if path already exists.

**Comment by kishen-v** [comment]
Job `pull-kubetest2-plugins-terraform-validate` had failed due to the changes in the asset's path. A supporting PR is raised - https://github.com/ppc64le-cloud/test-infra/pull/356.

File: data/unpack.go [hunk 2]
```diff
@@ -21,24 +31,27 @@ func Unpack(base string, uri string) (err error) {
...
+		files, err := dir.ReadDir(resPath)
+		if err != nil {
+			return fmt.Errorf("cannot read the provided directory - %v", err)
+		}
+		for _, file := range files {
+			name := file.Name()
+			err = Unpack(filepath.Join(extractPath, name), filepath.Join(resPath, name))
 			if err != nil {
 				return err
 			}
```
Comment (mkumatag) [inline data/unpack.go:49]:
```suggestion
			if err := Unpack(filepath.Join(extractPath, name), filepath.Join(resPath, name)); err != nil {
				return err
			}
```

**Review by mkumatag — CHANGES_REQUESTED** [review]
minor comments, overall lgtm

**Comment by kishen-v** [comment]
Changes post : https://github.com/ppc64le-cloud/kubetest2-plugins/pull/80

1. Update `Makefile` to clone the `k8s-submodule`.
2. Update Readme.md.

**Comment by mkumatag** [comment]
@Rajalakshmi-Girish Not sure why bot isn't merging, merging it manually.

**Pull request merged**